tests/nginx_test.go

```diff
@@ -0,0 +1,69 @@
+package bundles_test
+
+import (
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	manifestDir  = "/var/lib/rancher/k3s/server/manifests"
+	manifestFile = "ingress-nginx.yaml"
+)
+
+var _ = Describe("nginx test", Label("nginx"), func() {
+
+	BeforeEach(func() {
+		prepareBundle()
+	})
+	AfterEach(func() {
+		cleanBundle()
+	})
+
+	It("sets default version", func() {
+		runBundle()
+		dat, err := os.ReadFile(filepath.Join(manifestDir, manifestFile))
+		content := string(dat)
+		Expect(err).ToNot(HaveOccurred())
+		// renovate: depName=ingress-nginx repoUrl=https://kubernetes.github.io/ingress-nginx
+		Expect(content).To(ContainSubstring("version: \"4.7.3\""))
+	})
+
+	It("set version", func() {
+		err := os.WriteFile("/oem/foo.yaml", []byte(`#cloud-config
+nginx:
+ version: 1`), 0655)
+		Expect(err).ToNot(HaveOccurred())
+
+		runBundle()
+		dat, err := os.ReadFile(filepath.Join(manifestDir, manifestFile))
+		content := string(dat)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(content).To(ContainSubstring("version: \"1\""))
```

Comment:
`1` is handled differently to `4.2.7`:


```
$ docker run -v $PWD:/workdir --workdir /workdir  -it quay.io/kairos/core-alpine-opensuse-leap /bin/bash

0eb459861885:/workdir# vim /oem/foo.yaml

# Add the following config content and save

0eb459861885:/workdir# cat /oem/foo.yaml 
#cloud-config

foo:
  bar: 1
nginx:
  version: 4.2.7

0eb459861885:/workdir# kairos-agent config get "nginx.version | @json" 
INFO[2023-11-07T07:16:27Z] kairos-agent version v2.3.0                  
'"4.2.7"'
0eb459861885:/workdir# kairos-agent config get "foo.bar | @json" 
INFO[2023-11-07T07:16:31Z] kairos-agent version v2.3.0                  
"1"
```

`4.2.7` is returned double quoted while `1` is not.

Reply:
If `1` in the config above is changed to `4.2.7` (or any other version with dots in it), this test will fail.

Reply:
A simple solution would be to use `kairos-agent config get "nginx.version"` instead of `kairos-agent config get "nginx.version | @json"` since we don't expect `nginx.version` to be an object, it's just a string value. Without `@json`, the returned value is not quoted:

```
0eb459861885:/workdir# kairos-agent config get "nginx.version | @json" 
INFO[2023-11-07T07:32:17Z] kairos-agent version v2.3.0                  
'"4.2.7"'
0eb459861885:/workdir# kairos-agent config get "nginx.version" 
INFO[2023-11-07T07:32:19Z] kairos-agent version v2.3.0                  
4.2.7
```